src/host/services/IMUService.cpp

```diff
@@ -26,38 +26,204 @@
-  //DEBUG("Attitude roll: %f pitch: %f  Mag heading: %f", eulerAngles.z(), eulerAngles.y(), eulerAngles.x());
+  //DEBUG("Sensor Raw-datas: eulerAngles.z=%.3f eulerAngles.y=%.3f  eulerAngles.x=%.3f", eulerAngles.z(), eulerAngles.y(), eulerAngles.x());
 
   SKUpdateStatic<2> update;
   // Note: We could calculate yaw as the difference between the Magnetic
   // Heading and the Course Over Ground Magnetic.
 
-  /* if orientation == MOUNTED_ON_PORT_HULL */
-  double roll, pitch, heading;
-  roll = eulerAngles.z();
-  pitch = eulerAngles.y();
-  heading = fmod(eulerAngles.x() + 270, 360);
+  switch (_config.mounting) {
+    case VerticalPortHull:
+      _roll = SKDegToRad(eulerAngles.z());
+      _pitch = SKDegToRad(eulerAngles.y());
+      _heading = SKDegToRad(fmod(eulerAngles.x() + 270, 360));
+    break;
+    case VerticalStbHull:
+      _roll = SKDegToRad(eulerAngles.z())*(-1);
+      _pitch = SKDegToRad(eulerAngles.y())*(-1);
+      _heading = SKDegToRad(eulerAngles.x());
+    break;
+    case VerticalTopToBow:
+      _roll = SKDegToRad(eulerAngles.y());
+      _pitch = SKDegToRad(eulerAngles.z());
+      _heading = SKDegToRad(fmod(eulerAngles.x() + 180, 360));
+    break;
+    case HorizontalLeftSideToBow:
+      _roll = SKDegToRad(eulerAngles.y());
+      _pitch = SKDegToRad(eulerAngles.z()) * (-1);
+      _heading = SKDegToRad(eulerAngles.x());
+    break;
+  }
+
+  // update NavigationHeadingMagnetic if quality (= calibration value) is ok
+  if (isMagCalibrated()) {
+    update.setNavigationHeadingMagnetic(_heading);
+  }
+  // update NavigationAttitude if quality (= calibration value) is ok
+  if (isHeelAndPitchCalibrated()) {
+    update.setNavigationAttitude(SKTypeAttitude(/* roll */ _roll + _offsetRoll, /* pitch */ _pitch + _offsetPitch, /* yaw */ SKDoubleNAN));
+    //DEBUG("Heel = %.3f | Pitch = %.3f | Heading = %.3f", SKRadToDeg( _roll + _offsetRoll), SKRadToDeg( _pitch + _offsetPitch), SKRadToDeg(_heading));
+  }
+  _skHub.publish(update);
+
+  // Save calibration values to EEPROM every 30 Minutes, if BNO055 is fully calibrated
+  if (bno055.isFullyCalibrated() && _timeSinceLastCalSave > 1800000) {
+    if (saveCalibrationOffsets())
+      _timeSinceLastCalSave = 0;
+  }
+}
+
+// get the actual values of calibration, heel & pitch (including offset) and heading for display
+// all angles in radians
+void IMUService::getLastValues(int &sysCalibration, int &accelCalibration, double &pitch, double &roll, int &magCalibration, double &heading){
+  sysCalibration = _sysCalib;
+  accelCalibration = _accelCalib;
+  pitch = _pitch + _offsetPitch;
+  roll = _roll + _offsetRoll;
+  magCalibration = _magCalib;
+  heading = _heading;
+}
+
+//  Calc offset for making Heel=0 and Pitch=0
+//  (e.g. called with long button press in IMUMonitorPage)
+//  If an setOffset will be done a second time within 5 Seconds,
+//  the values will be stored into EEPROM and loaded at start of KBox
+void IMUService::setHeelPitchOffset() {
```

Comment:
Why do you use `heel` sometimes and `roll` some other times? I think it's a bit confusing and we should probably use only one.

Reply:
The problem is that e.g. openCPN or any display shows "heel" and not "roll"
And also when we come to performance calculations everywhere is mentioned "heel"

Remember, when I asked you to change the wording to "heel"?
It is a pity, that SignalK has chosen "roll"....

So I took "heel" as much as possible and let roll where it has to be (IMU sensor, SignalK)

Reply:
This is also what NMEA2000 is using: https://github.com/ttlappalainen/NMEA2000/blob/master/src/N2kMessages.h#L403

It's ok to use heel on the display but I do not think we should use it in the code. Having two names for the same thing is just confusing and I am tempted to follow what everyone else is doing here.